### ISSUE TITLE
Programmatic (non- Interface Builder) UISplitViewController apps cannot initialize.

### ISSUE THREAD

**Issue by gary17** [Microsoft/WinObjC#1128]
According to Apple documentation:

> You can configure the [UISplitViewController] child view controllers using Interface Builder or programmatically by assigning the view controllers to the viewControllers property

However, the `viewControllers` property is reported as not implemented:

`(...) warning : 'viewControllers' is deprecated: property not yet implemented [-Wdeprecated-declarations]`

Assigning an array of child view controllers to a split view controller results in `UIApplicationDelegate` launching properly but view controller `[viewDidLoad]` never getting called.

Effectively, programmatic (non- Interface Builder) `UISplitViewController` apps cannot initialize.

The problem will be similar for other UI controls that *initialize* through the `viewControllers` property (such as `UITabBarController`), as opposed to *exposing* already initialized child view controllers (such as `UINavigationController`).

The problem seems to be analogous to [issue #411](http://github.com/Microsoft/WinObjC/issues/411).

As of: WinObjC version 0.2.160927 (September 27, 2016).

**Comment by ArnOmsft** [comment]
Thanks for the feedback. We won't be able to address this in the short term but you should consider submitting a pull request for this if you'd like to contribute to the project.

**Comment by gary17** [comment]
Converted a basic Xcode 8 "Master-Detail Application" project template to WinObjC using `vsimporter.exe`, but `[application:didFinishLaunchingWithOptions:]` fails to receive an instance of `UISplitViewController`:

```objective-c
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
	// always nil
	UISplitViewController *splitViewController = (UISplitViewController *)self.window.rootViewController;

	UINavigationController *navigationController = [splitViewController.viewControllers lastObject];

	navigationController.topViewController.navigationItem.leftBarButtonItem = splitViewController.displayModeButtonItem;
	splitViewController.delegate = self;

	return YES;
}
```

Considering that 1.) `UISplitViewController` is probably the most commonly used UI control for tablet-style (iPad) apps, especially in the Business and Productivity areas, and 2.) Microsoft Surface tablets and laptop/tablet 2-in-1s are the strongest Windows 10 mobile devices - is `UISplitViewController` planned to be supported?

Alternatively, is there perhaps a way to create a split view controller through XAML and hook `UINavigationController` instances into it?

**Comment by ArnOmsft** [comment]
Sorry for the late answer. Supporting UISplitViewController is currently not a priority for us.
@jaredhms, can you help answer @gary17 question: "_is there perhaps a way to create a split view controller through XAML and hook UINavigationController instances into it?_"
Thanks.

**Comment by gary17** [comment]
Yes, I'm still interested whether it is possible to create a UWP split view control programmatically or through XAML and hook `UINavigationController` instances into it.

**Comment by gary17** [comment]
Any news on `UISplitViewController` support or perhaps a way to create a split view controller through XAML and hook it up manually?
